core/plugin.go

```diff
@@ -95,21 +99,56 @@ type SubscribedPlugin interface {
 }
 
 type RequestedPlugin struct {
-	path       string
-	checkSum   [sha256.Size]byte
-	signature  []byte
-	autoLoaded bool
-}
-
-func NewRequestedPlugin(path string) (*RequestedPlugin, error) {
-	rp := &RequestedPlugin{
-		path:       path,
-		signature:  nil,
-		autoLoaded: true,
-	}
-	err := rp.generateCheckSum()
-	if err != nil {
-		return nil, err
+	path      string
+	checkSum  [sha256.Size]byte
+	signature []byte
+}
+
+func NewRequestedPlugin(path, tmp string, b []byte) (*RequestedPlugin, error) {
```

Comment:
Can we add a comment about the expected behavior of this? Especially with b being nil vs not being nil.